Tools/Scripts/webkitdirs.pm

```diff
@@ -730,12 +730,8 @@ sub determineNumberOfCPUs
     if (defined($ENV{NUMBER_OF_PROCESSORS})) {
         $numberOfCPUs = $ENV{NUMBER_OF_PROCESSORS};
     } elsif (isLinux()) {
-        # First try the nproc utility, if it exists. If we get no
-        # results fall back to just interpretting /proc directly.
-        chomp($numberOfCPUs = `nproc --all 2> /dev/null`);
-        if ($numberOfCPUs eq "") {
-            $numberOfCPUs = (grep /processor/, `cat /proc/cpuinfo`);
-        }
+        # Don't use nproc because it returns unreliable results, https://webkit.org/b/273675
+        $numberOfCPUs = (grep /processor/, `cat /proc/cpuinfo`);
```

Comment:
I still don't like this. And it is not really needed to shell out to `nproc`, or even to `grep`... remember that Perl is supposed to be a glorified `grep`. So I would instead suggest:

```suggestion
        use POSIX;
        $numberOfCPUs = POSIXL::sysconf(83); # _SC_NPROCESSORS_ONLN = 83
        if ($numberOfCPUs == "") {
            open CPUINFO, "/proc/cpuinfo";
            $numberOfCPUs = 0;
            while (<CPUINFO>) {
                if (/^[Pp]rocessor\s/) { $numberOfCPUs++; }
            }
            close CPUINFO;
        }
```

(Yes, I tested both the `sysconf` code path and the fallback to parsing `/proc/cpuinfo` here.)

Reply:
OK, but now you've written 100% of the code in this pull request, so we might as well close this one and let you open a new one for me to review?

Reply:
Done: https://github.com/WebKit/WebKit/pull/28645